patches/server/0910-Add-WardenAngerEvent.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: nopjar <code.nopjar@gmail.com>
+Date: Sun, 12 Jun 2022 02:26:04 +0200
+Subject: [PATCH] Add WardenAngerEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/warden/Warden.java b/src/main/java/net/minecraft/world/entity/monster/warden/Warden.java
+index 60410fd12e0fcfbaad3b4dc431452f8b2e6276c7..11e09165bca185ccfaa96ddcd62856e694e6a0f4 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/warden/Warden.java
++++ b/src/main/java/net/minecraft/world/entity/monster/warden/Warden.java
+@@ -361,7 +361,7 @@ public class Warden extends Monster implements VibrationListener.VibrationListen
+ 
+     @Override
+     public Brain<Warden> getBrain() {
+-        return super.getBrain();
++        return (Brain<Warden>) super.getBrain(); // Paper - decompile fix
+     }
+ 
+     @Override
+@@ -460,9 +460,17 @@ public class Warden extends Monster implements VibrationListener.VibrationListen
+     @VisibleForTesting
+     public void increaseAngerAt(@Nullable Entity entity, int amount, boolean listening) {
+         if (!this.isNoAi() && this.canTargetEntity(entity)) {
++            // Paper start
++            io.papermc.paper.event.entity.WardenAngerEvent event = new io.papermc.paper.event.entity.WardenAngerEvent((org.bukkit.entity.Warden) this.getBukkitEntity(), entity.getBukkitEntity(), this.angerManagement.getActiveAnger(entity) + amount);
++            this.level.getCraftServer().getPluginManager().callEvent(event);
++            if (event.isCancelled()) {
++                this.angerManagement.clearAnger(entity);
```

Comment:
Could just be me but why clear it's angry? Why not just return? The WardenAngerEvent fires each time the Wardens anger value increases if we wanted to clear it wouldn't we just want to use setNewAnger(0) instead?

Reply:
We already talked about that in #paper-contrib. Noah pointed out that, for the current layout of the event, it does make sense. The event only represents the warden being angered so you would expect cancellation to not have an angry warden.

Maybe it is worth actually moving this event to a WardenChangeAngerEvent (or something of the likes) ?
Would also make the fact that it fires more often a bit clearer.

Reply:
I think a name change is best since the Warden can be "angry" but not actually do anything under 80 angry value.

Reply:
Jap, makes much more sense. I'll change it